packages/compass-components/src/components/drawer-portal.tsx

```diff
@@ -0,0 +1,307 @@
+  // so we override the values to hide the drawer toolbar when there's nothing
+  // to show
+  '& > div:nth-child(2)': {
+    width: '0 !important',
+    overflow: 'hidden',
+  },
+});
+
+const drawerSectionPortalStyles = css({
+  minWidth: '100%',
+  minHeight: '100%',
+});
+
+/**
+ * DrawerAnchor component will render the drawer in any place it is rendered.
+ * This component has to wrap any content that Drawer will be shown near
+ */
+export const DrawerAnchor: React.FunctionComponent<{
+  displayMode?: DrawerDisplayMode;
+}> = ({ displayMode, children }) => {
+  const actions = useContext(DrawerActionsContext);
+  const drawerSectionItems = useContext(DrawerStateContext);
+  const prevDrawerSectionItems = useRef<DrawerSectionProps[]>([]);
+  useEffect(() => {
+    const prevIds = new Set(
+      prevDrawerSectionItems.current.map((data) => {
+        return data.id;
+      })
+    );
+    for (const item of drawerSectionItems) {
+      if (!prevIds.has(item.id) && item.autoOpen) {
+        rafraf(() => {
+          actions.current.openDrawer(item.id);
+        });
+      }
+    }
+    prevDrawerSectionItems.current = drawerSectionItems;
+  }, [actions, drawerSectionItems]);
+  const toolbarData = useMemo(() => {
+    return drawerSectionItems
+      .map((data) => {
+        return {
+          ...data,
+          content: (
+            <div
+              data-drawer-section={data.id}
+              className={drawerSectionPortalStyles}
+            ></div>
+          ),
+        };
+      })
+      .sort(({ order: orderA = Infinity }, { order: orderB = Infinity }) => {
+        return orderB < orderA ? 1 : orderB > orderA ? -1 : 0;
+      });
+  }, [drawerSectionItems]);
+  return (
+    <DrawerLayout
+      displayMode={displayMode ?? DrawerDisplayMode.Embedded}
+      toolbarData={toolbarData}
+      className={cx(
+        drawerLayoutFixesStyles,
+        toolbarData.length === 0 && emptyDrawerLayoutFixesStyles,
+        // classname is the only property leafygreen passes over to the drawer
+        // wrapper component that would allow us to target it
+        'compass-drawer-anchor'
+      )}
+    >
+      <DrawerContextGrabber>{children}</DrawerContextGrabber>
+    </DrawerLayout>
+  );
+};
+
+/**
+ * DrawerSection allows to declaratively render sections inside the drawer
+ * independantly from the Drawer itself
+ */
+export const DrawerSection: React.FunctionComponent<DrawerSectionProps> = ({
+  children,
+  ...props
+}) => {
+  const [portalNode, setPortalNode] = useState<Element | null>(null);
+  const actions = useContext(DrawerActionsContext);
+  const prevProps = useRef<DrawerSectionProps>();
+  useEffect(() => {
+    if (!isEqual(prevProps.current, props)) {
+      actions.current.updateToolbarData({ autoOpen: false, ...props });
+      prevProps.current = props;
+    }
+  });
+  useLayoutEffect(() => {
+    const drawerEl = document.querySelector(
+      '.compass-drawer-anchor > div:nth-child(2)'
```

Comment:
Would be really nice if we could get some id or something to target this in a way that won't be so brittle?

Reply:
Yeah, I'll be reaching out to lg team about this